### PR TITLE
Added dynamic layer support.

### PR DESCRIPTION
**Pull request by teamagam3**
Added new router to dynamic layer, that containts the following routes:
* POST new dynamic layer
* POST /:id add entity to existing dynamic layer
* DELETE /:layerId/:entityId delete entity from existing layer

### CODE REVIEW

File: routes/common.js [hunk 1]
```diff
@@ -235,7 +235,7 @@ function removeIconById(id, onFailure, onSuccess) {
         if (err) {
             return onFailure(err);
         }
-        return onSuccess(oldDoc);
+        return onSuccess(newIcon);
```
Comment (BarAgam) [inline routes/common.js:235]: 😨 
You didn't rename the function's parameter

File: routes/dynamic-layers.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        common.saveNewMessageToMongoDB(layer,
+            function (err) {
+                error(err);
+            }, function (doc) {
+                success(doc);
+            });
+    });
+}
+
+function addIdToEntity(entity) {
```
Comment (BarAgam) [inline routes/dynamic-layers.js]: Should be before findByIdAndSend

File: routes/dynamic-layers.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        });
+    }, (layerToSend) => res.send(layerToSend), next);
+});
+
+function findByIdAndSend(layerId, success, error) {
+    findByIdAndSend(layerId, undefined, success, error);
+}
+
+function findByIdAndSend(layerId, changeLayer, success, error) {
+    messageModel.findById(layerId, function (err, layer) {
```
Comment (BarAgam) [inline routes/dynamic-layers.js]: There are going to be multiple messages of the same layer with different time-stamps, for each update the layer gets.
Will this return the latest version of the layer ?

File: routes/dynamic-layers.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    });
+}
+
+function addIdToEntity(entity) {
+    entity.id = uuid();
+
+    return entity;
+}
+
+function getAsNewMessage(layer) {
```
Comment (BarAgam) [inline routes/dynamic-layers.js]: This name is kinda confusing....

**Review by BarAgam — CHANGES_REQUESTED** [review]

File: routes/dynamic-layers.js (hunk 3 above)
Comment (teamagam3) [reply routes/dynamic-layers.js]: Yeah, each one of the layers has unique id, that means the the client should update itself each request.
I should change it to consistent id for each layer, then I'll change the `findById` to `findOne` and sort by date.

**Pull request merged**